airflow/dag_processing/processor.py

```diff
@@ -412,20 +411,19 @@ def manage_slas(self, dag: DAG, session: Session = None) -> None:
             else:
                 while next_info.logical_date < ts:
                     next_info = dag.next_dagrun_info(next_info.data_interval, restricted=False)
-
                     if next_info is None:
                         break
-                    if (ti.dag_id, ti.task_id, next_info.logical_date) in recorded_slas_query:
+                    if (ti.dag_id, ti.task_id, ti.run_id, ti.map_index) in recorded_sla_misses:
```

Comment:
Hmm, this looks suspecting. The previous logic checks if the *next* logical date is in recorded SLA misses, while your new logic checks _this_ run ID.

Reply:
Yeah true I'll take another look. Maybe it needs another test

Reply:
Go figure -- it was too much to hope that this change would be nice and simple, eh!

Reply:
yeah i had it that way originally, and the tests were failing due to referential integrity errors.  and the reason is because the TI did not exist.  but that is expected with SLAs.  so we can't add a FK for SLA->TI unless we make `manage_slas` _also_ create the TI.

For now [here's a PR](https://github.com/apache/airflow/pull/22214) that makes an sla test a little stricter to document *which* sla miss is created, and not just *that* an sla miss is created.

And I guess for now I'll remove the FK and update the code.

Reply:
hmm... but this also presents a problem in the context of mapping. 

consider this logic:

```python
                    if (ti.dag_id, ti.task_id, next_run_id, ti.map_index) in recorded_sla_misses:
                        break
                    if next_info.logical_date + task.sla < ts:
                        sla_misses.append(
                            SlaMiss(
                                task_id=ti.task_id,
                                dag_id=ti.dag_id,
                                run_id=next_run_id,
                                timestamp=ts,
                            )
                        )
```

In the context of mapping, it's not really reasonable to expect that a TI with this particular map index should exist in the next run -- that's kindof the point of mapping, you don't know what tasks will need to run from one run to the next.  

It feels like we need to keep SlaMiss at the grain of dag / run / task, and that mapped tasks should really be thought of as more like a subtask in this context.  Indeed maybe we should consider making that more formal.  WDYT?

Reply:
Hmmm, SLAs are horribly broken in many ways right now, and I'd like us to not get into the weeds here fixing that _now_.

Thinking what is the minimum we can do. Like I wonder if we could not touch _anything_ here, and (for now/the 2.3 release) we forbid SLAs form being set on mapped tasks?

Reply:
Yeah I reckon we could do that

Reply:
did the work to ignore in manage_slas (i think)

may need help disabling sla for mappped tasks